### PR TITLE
add support for custom environment variables to the server deployment

### PR DESCRIPTION
**Pull request by andrewrothstein**
a corporate L7 proxy sits between my kiali deployment and my prometheus server. I need to specify the (HTTP|HTTPS|NO)_PROXY environment variables in my helm chart deployment.

part of: https://github.com/kiali/kiali/issues/7569

### CODE REVIEW

**Comment by jmazzitelli** [comment]
Please review the [community contribution document](https://github.com/kiali/kiali/blob/master/CONTRIBUTING.md#making-a-change) for the process that is in place before submitting any PRs. We ask that a github issue be created first which describes the details of the bug or enhancement request being asked for. The issue provides a chance for the Kiali maintainers to have a dialogue with you before you submit any PRs.

For example, this is not complete without an accompanying operator PR.

In addition, I believe this request has been asked for before - though I can't remember the resolution (if there was one). Did you search in the github issues/discussions for a request for proxy settings?

**Review by jmazzitelli — DISMISSED** [review]
there is no accompanying operator PR that goes with this server helm chart PR. We cannot merge this without that additional work. See checklists:
* https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-configuration-setting
* https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-resource

**Comment by jmazzitelli** [comment]
see: https://github.com/kiali/kiali/issues/6386

**Comment by jmazzitelli** [comment]
This PR looks correct as-is. Since we need feature parity with the operator, here's other places to touch for the new [operator PR](https://github.com/kiali/kiali-operator/pulls) we need:

* operator defaults: https://github.com/kiali/kiali-operator/blob/v1.89.7/roles/default/kiali-deploy/defaults/main.yml#L72
* CRD Schema (for docs and validator): https://github.com/kiali/kiali-operator/blob/v1.89.7/crd-docs/crd/kiali.io_kialis.yaml#L305
* CR example (for docs): https://github.com/kiali/kiali-operator/blob/v1.89.7/crd-docs/cr/kiali.io_v1alpha1_kiali.yaml#L111
* non-OpenShift Kubernetes deployment template: https://github.com/kiali/kiali-operator/blob/v1.89.7/roles/default/kiali-deploy/templates/kubernetes/deployment.yaml#L116
* OpenShift deployment template: https://github.com/kiali/kiali-operator/blob/v1.89.7/roles/default/kiali-deploy/templates/openshift/deployment.yaml#L116
* Molecule test update (config-values-test):
  * add some custom env vars similar to the way custom secrets are done: https://github.com/kiali/kiali-operator/blob/v1.89.7/molecule/config-values-test/converge.yml#L157-L167
  * check that the custom env vars made it to the pod spec, similar to how the securityContext is checked: https://github.com/kiali/kiali-operator/blob/v1.89.7/molecule/config-values-test/converge.yml#L324-L333

**Comment by jmazzitelli** [comment]
I'm going to work on the operator implementation. Should be done today. This is going to be done as part of https://github.com/kiali/kiali/issues/7569

**Comment by jmazzitelli** [comment]
Test procedures for this PR:

1. Build the helm charts: `make build-helm-charts`
2. Run "helm template" and set some custom envs - confirm the envs are being set in the deployment, ensuring things like booleans and numbers are properly quoted:
```sh
helm template \
  --set deployment.custom_envs[0].name="FOO1" \
  --set deployment.custom_envs[0].value="http://aa.example.com:123" \
  --set deployment.custom_envs[1].name="FOO2" \
  --set deployment.custom_envs[1].value=123 \
  --set deployment.custom_envs[2].name="FOO3" \
  --set deployment.custom_envs[2].value=true \
  -n istio-system kiali-server _output/charts/kiali-server-*.tgz
```
should result in the following within the Deployment:
```yaml
        - name: "FOO1"
          value: "http://aa.example.com:123"
        - name: "FOO2"
          value: "123"
        - name: "FOO3"
          value: "true"
```

**Comment by jmazzitelli** [comment]
operator PR that introduces this functionality: https://github.com/kiali/kiali-operator/pull/829

File: kiali-server/values.yaml
Comment (jmazzitelli) [inline kiali-server/values.yaml:42]: @andrewrothstein please delete this comment. It is superfluous and just adds more clutter to the list of settings here. This is the only thing that needs to change in this PR. Once this is done and I get the operator PR reviewed, this PR and the operator PR can be merged.
```suggestion
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Review by jshaughn — DISMISSED** [review]
Ran through the test steps and it worked.  I'll approve and merge (assuming the operator part works as well).  We can remove the bad comment iin a trivial follow-up, not sure the original author will get to it.

**Comment by jmazzitelli** [comment]
> We can remove the bad comment in a trivial follow-up

I was able to commit it - github allows maintainers to commit suggestions so I had permission from the github UI to do it.

**Review by jmazzitelli — APPROVED** [review]

**Pull request merged**
